_posts/tutorials/2021-04-16-javalin-sureness-example.md

```diff
@@ -0,0 +1,298 @@
+---
+layout: tutorial
+title: "Using Sureness to protect the security of Javalin REST API"
+author: <a href="https://github.com/tomsun28" target="_blank">Tom</a>
+date: 2021-04-16
+summarytitle: Using Sureness to protect the security of Javalin REST API.
+summary: Using Sureness to secure Javalin REST API by providing authentication(jwt,bsaic,digest) and authorization(rbac)"
+language: Java
+---
+
+## What is Sureness   
+
+> [Sureness](https://github.com/usthe/sureness) is a simple and efficient open-source security framework that focus on protection of restful api.  
+> Provide authentication and authorization, based on RBAC.   
+> No specific framework dependency(support javalin, springboot, quarkus, ktor and more).    
+> Support dynamic modification of permissions.   
+> Support websocket, mainstream http container(servlet and jax-rs).    
+> Supports JWT, Basic Auth, Digest Auth... Can extend custom supported authentication methods.    
+> High performance due dictionary matching tree.      
+> Good extension interface, demo and document.    
+
+> The low configuration of sureness, easy to expand, and not coupled with other frameworks, enables developers to quickly and safely protect their projects in multiple scenarios.   
+
+
+## What You Will Learn
+
+* Creating a simple REST API using Javalin  
+* Learn how to integrate Sureness into a Javalin application  
+* Learn how to issue a jwt  
+* Test API authentication - use jwt auth, basic auth, digest auth to test the security of the REST API   
+* Test API authorization - use different users to verify that they can access the REST API   
+
+
+The tutorial assumes that you know what jwt, basic auth, digest auth, rbac are. If you 
+do not, then you can check [jwt](https://jwt.io/introduction/), [basic auth](https://docs.oracle.com/cd/E50612_01/doc.11122/user_guide/content/authn_http_basic.html) , [digest auth](https://docs.oracle.com/cd/E50612_01/doc.11122/user_guide/content/authn_http_digest.html), [rbac](https://en.wikipedia.org/wiki/Role-based_access_control) for an introduction.
+
+## Setting Up Dependencies
+
+First, you will need to create a maven project and add javalin, sureness dependencies coordinate   
+
+````xml
+        <dependency>
+            <groupId>io.javalin</groupId>
+            <artifactId>javalin</artifactId>
+            <version>3.10.1</version>
```

Comment:
The latest version is `3.13.6`, but you can use `{{site.javalinversion}}`, and it will update itself automatically:

```
<version>{{site.javalinversion}}</version>
```